server/src/main/java/org/elasticsearch/common/io/stream/StreamOutput.java

```diff
@@ -760,6 +762,24 @@ public final void writeOptionalInstant(@Nullable Instant instant) throws IOExcep
             o.writeInt(period.getYears());
             o.writeInt(period.getMonths());
             o.writeInt(period.getDays());
+        }),
+        entry(GenericNamedWriteable.class, (o, v) -> {
+            // Note that we do not rely on the checks in VersionCheckingStreamOutput because that only applies to CCS
+            final var genericNamedWriteable = (GenericNamedWriteable) v;
+            TransportVersion minSupportedVersion = genericNamedWriteable.getMinimalSupportedVersion();
+            assert minSupportedVersion.onOrAfter(V_8_500_070) : "[GenericNamedWriteable] requires [" + V_8_500_070 + "]";
```

Comment:
Might be worth mentioning `minSupportedVersion` in the message too.

Reply:
Since this was in an assert that would only happen in tests, and therefor tracked by a developer who will easily follow the code, I favoured brevity (and a non-wrapping line) over detail.

The next line that does the real check for production code is much more informative (for both the user and any developer tracking a failure).

Reply:
Makes sense, thanks.